vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java

```diff
@@ -219,7 +222,12 @@ private void initBaseElementsAndListeners() {
     private void removeAutoAdded() {
         if (autoAddedToTheUi && !isOpened()) {
             autoAddedToTheUi = false;
```

Comment:
If you refresh the page while an auto-added notification (`autoAddedToTheUi`) is open, the flag gets cleared but the notification element doesn't get removed from its parent (because of the original issue) so the notification is no longer considered auto-added!

Add the following test case:
1. open the notification
2. refresh the page
3. open the notification (if not already open)
4. close the notification
5. check that the notification was removed from its parent container

To fix the issue, set this flag `false` _only_ if the element gets successfully removed from its parent by the logic below.